### PR TITLE
feat(art-quiz) create triggerCampaignMutation and gravity loader

### PR DESCRIPTION
**Pull request by TMcMeans**
The type of this PR is: feat

This PR solves [GRO-1262]

### Description
The goal of this PR is to add a mutation for a `POST` request that triggers a Braze campaign send for authenticated users taking the art quiz to receive their results via email. 

This mutation is also pretty generalized so that any client making a call to `/me/trigger_campaign` with a new campaign can be mapped w/ an appropriate `campaign_id`. A spec is included to test that the call was made w/ the correct values passed to gravity. 

TODO:
- [x] talk to marketing to replace the current `campaign_id` for art quiz to the correct value from Braze
- [ ] make the mutation call in Force on the art quiz "email me my results" button click w/ the correct `campaign_id`

Note: the `clientMutationId` is returned as a response and returns null but that seems to be a graphql/relay thing and isn't a problem here.

Screen Shot 2022-12-22 at 2 43 37 PM

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[GRO-1262]: https://artsyproduct.atlassian.net/browse/GRO-1262?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/schema/v2/me/triggerCampaignMutation.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  outputFields: {},
+
+  mutateAndGetPayload: async (args, { triggerCampaignLoader }) => {
+    if (!triggerCampaignLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    return triggerCampaignLoader({ campaign_id: args.campaignID })
+  },
+})
```
Comment (damassi) [inline src/schema/v2/me/triggerCampaignMutation.ts:37]: @dzucconi - Matt Z has been flagging me on the ThingOrError pattern lately with new mutations. Thoughts?

Comment (dzucconi) [reply src/schema/v2/me/triggerCampaignMutation.ts:37]: I dislike it a great deal, personally. If it was a good idea it should be part of the spec. But [here's an example](https://github.com/artsy/metaphysics/blob/main/src/schema/v2/me/mark_all_notifications_as_read_mutation.ts) @TMcMeans

Comment (TMcMeans) [reply src/schema/v2/me/triggerCampaignMutation.ts:37]: I dont have an opinion either way just wanna make sure I'm sticking to convention, thanks for the example

Comment (TMcMeans) [reply src/schema/v2/me/triggerCampaignMutation.ts:37]: love this.... but whats ChatGPT??

Comment (damassi) [reply src/schema/v2/me/triggerCampaignMutation.ts:37]: (I feel weird suggesting this, but this is exactly what ChatGPT is good at, and I just did this for like 7 mutations and it worked great lol)

Comment (TMcMeans) [reply src/schema/v2/me/triggerCampaignMutation.ts:37]: @damassi https://openai.com/blog/chatgpt/??

Comment (damassi) [reply src/schema/v2/me/triggerCampaignMutation.ts:37]: https://chat.openai.com

**Comment by damassi** [comment]
Sorry, just deleted my annoying comment / thread about ChatGPT -- Should probably learn how to type this out by hand before bots start writing it!

**Comment by TMcMeans** [comment]
other than any edits based on comments I'm just waiting to confirm the correct Braze `campaign_id` to move this pr out of draft status.

File: src/schema/v2/me/triggerCampaignMutation.ts (hunk 1 above)
Comment (damassi) [reply src/schema/v2/me/triggerCampaignMutation.ts:112]: Looks good!

**Review by laurabeth — APPROVED** [review]
LGTM. :shipit:

**Pull request merged**